### DISCUSSION TITLE
[Question] Use-cases for HTTP Handler methods

### DISCUSSION THREAD

**Discussion by Cheddah** [uNetworking/uWebSockets.js#229] [Q&A] [answered]
Hello,

This may be due to my lack of understanding of how WebSockets work, but I thought it wouldn't hurt to ask: why does one need TemplatedApp methods outside of `TemplatedApp::ws`? To my understanding, once you use `ws`, you've successfully established a websocket connection, and the client wouldn't (or shouldn't) be sending HTTP requests to your server. So wouldn't the event handlers that you specify (via `WebSocketBehavior`) be all you need?

In other words, **if you're relying on HTTP method handlers, doesn't that imply that you're not using WebSockets**? Please let me know if my understanding is correct.

Thanks in advance :)

**Comment by Cheddah** [comment]
Bump

I don't mean to come off as facetious; I just want to verify my understanding of websockets with how this library should work 👍

**Comment by Cheddah** [comment, marked as answer]
The answer was in the first sentence of the README that I glossed over:

> Think of it as a complete replacement to both Express.js...

With that context in mind, I understand now.
